plugin/storage/kafka/marshaller.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"github.com/apache/thrift/lib/go/thrift"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/model/converter/thrift/jaeger"
+)
+
+// Marshaller encodes a span into a byte array to be sent to Kafka
+type Marshaller interface {
+	Marshal(*model.Span) ([]byte, error)
+}
+
+type thriftMarshaller struct {
+	memBuffer *thrift.TMemoryBuffer
+	protocol  *thrift.TBinaryProtocol
+}
+
+func newThriftMarshaller() *thriftMarshaller {
+	memBuffer := thrift.NewTMemoryBuffer()
+	return &thriftMarshaller{
+		memBuffer: memBuffer,
+		protocol:  thrift.NewTBinaryProtocolTransport(memBuffer),
+	}
+}
+
+// Marshall encodes a span as a thrift byte array
+func (h *thriftMarshaller) Marshal(span *model.Span) ([]byte, error) {
+	thriftSpan := jaeger.FromDomainSpan(span)
+
+	thriftSpan.Write(h.protocol)
+
+	return h.memBuffer.Bytes(), nil
```

Comment:
- does membuffer need to be reset between writes?
- can this be called concurrently?